GENIE v3 Integration

jminock opened the pull request:
Updated Dockerfile to install necessary dependencies
Updated WCSim Cmake to use correct GENIEv3 libraries and find GENIE header directory
Removed NO_GENIE flag
Updated header paths for GENIE header files
Updated sourceme for new dependencies

jminock committed:
Update to GENIE v3

jminock committed:
Merge pull request #1 from ANNIEsoft/annie

Remove deprecated CommitHash.txt from CMakeLists

jminock committed:
Update to GENIE v3

jminock committed:
Update to GENIE v3

jminock committed:
Update to GENIE v3

jminock committed:
Update path to GENIE headers and remove NO_GENIE

jminock committed:
Update to include GENIE v3

marc1uk commented:
Hey james. Looks good, just giving it a cursory test run. 
Is there a reason you removed the NO_GENIE flag? Genie+pythia+log4cpp+LHAPDF amount to quite a lot of bloat if you're not actually interested in using GENIE primaries. Some folks might just want to simulate muons (to simulate LAPPD response), or neutrons (to investigate neutron capture response), in which case this is a lot of extra baggage they don't need. The NO_GENIE flag let users just get WCSim with its minimal dependencies of GEANT4 and ROOT, and if it wasn't set, then it had no effect. Not sure why you'd remove that.

jminock commented:
Hey Marcus, I'm sorry I wasn't thinking of that! I removed it because regardless of how the NO_GENIE flag was set (0/1 or true or false) the code would always treat it as defined and not read in GENIE events. Given I wasn't thinking of use cases that don't require GENIE, I thought it would be easier to just get rid of it. So I think a quick fix is to just include some documentation to unset NO_GENIE if one plans on using GENIE and I'll put the flag back in for a new PR, sound good?

jminock committed:
Added NO_GENIE back

jminock committed:
Enabled devtoolset-3

marc1uk committed:
Update Dockerfile

some new dependency must have introduced a system install of g++, so we can no longer simply symlink in g++/gcc/etc to `/usr/bin/`.
Instead use the devtoolset enable script to set up the development environment.
